fedimint-core/src/db/notifications.rs

```diff
@@ -0,0 +1,222 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+
+use async_trait::async_trait;
+use bitvec::vec::BitVec;
+use tokio::sync::futures::Notified;
+use tokio::sync::Notify;
+
+use super::{IDatabaseTransaction, PrefixStream, Result};
+
+/// Number of buckets used for `Notifications`.
+const NOTIFY_BUCKETS: usize = 32;
```

Comment:
How is 32 chosen?  Should it depend on the number of connected clients?

Reply:
just arbitrary

Reply:
It's memory usage vs spurious wakeups tradeoff.

Reply:
we can have it dynamic change based on number on notifications.